### DISCUSSION TITLE
✅  Unable to access devices on local network

### DISCUSSION THREAD

**Discussion by RonaldTHerman** [isontheline/pro.webssh.net#520] [Q&A] [answered]
I get a “connection failed” when attempting to WebSSH connect to a raspberry pi Linux box. This used to work. If I do the same from my macOS it works. I can also WebSSH connect to the macOS. Don’t know what changed. Suggestions? Any way to see a log?

**Comment by isontheline** [comment]
Hello @RonaldTHerman 👋

In order to enable verbose logging you can follow the steps :
1. iOS Settings
2. WebSSH
3. Settings
4. Log Level > Verbose
5. Restart WebSSH
6. Reproduce your action
7. Find the log files inside iOS > Files > WebSSH

**Comment by RonaldTHerman** [comment]
What does “no route to host” mean?







---------------------------------
Ronald T. Herman
***@***.***
Canterbury NH

> On Jan 24, 2022, at 12:29 AM, Arnaud Mengus ***@***.***> wrote:
> 
> ﻿
> Hello @RonaldTHerman 👋
> 
> In order to enable verbose logging you can follow the steps :
> 
> iOS Settings
> WebSSH
> Settings
> Log Level > Verbose
> Restart WebSSH
> Reproduce your action
> Find the log files inside iOS > Files > WebSSH
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android. 
> You are receiving this because you were mentioned.

2022/01/24 00:48:02:113 [VERBOSE] No prefered language. Use system language if available.
2022/01/24 00:48:02:130 [VERBOSE] Locale Identifier : en_US
2022/01/24 00:48:02:130 [VERBOSE] Device language : en
2022/01/24 00:48:02:131 [VERBOSE] Selected language : en
2022/01/24 00:48:02:639 [VERBOSE] Did become active [AppDelegate]
2022/01/24 00:48:02:644 [VERBOSE] No pin code set. WebSSH won't be locked.
2022/01/24 00:48:03:345 [DEBUG] Launch_APP_Count_16.3 : 7
2022/01/24 00:48:03:348 [DEBUG] Launch_APP_Count : 34
2022/01/24 00:48:03:397 [VERBOSE] IAP already purchased : com.webssh.pro.unlimited
2022/01/24 00:48:03:530 [VERBOSE] Updating host statuses...
2022/01/24 00:48:04:535 [VERBOSE] Host 192.168.86.67:22 changed. Is up : 0
2022/01/24 00:48:05:114 [VERBOSE] Launching connection...
2022/01/24 00:48:05:117 [VERBOSE] Connecting to SSH...
2022/01/24 00:48:05:117 [VERBOSE] Launching connection with block...
2022/01/24 00:48:05:121 [VERBOSE] Won't dismiss prompt because no one is displayed.
2022/01/24 00:48:05:236 [VERBOSE] <WebSSHLog>
  <Connect_Ssh>
    <WebSSHVersion>9.5.0.88</WebSSHVersion>
    <connectInner>
      <hostname>192.168.86.152</hostname>
      <port>22</port>
      <sshConnect>
        <connectSocket>
          <connect_ipv6_or_ipv4>
            <socketErrno>65</socketErrno>
            <socketError>No route to host</socketError>
          </connect_ipv6_or_ipv4>
          <connect_ipv6_or_ipv4>
            <socketErrno>65</socketErrno>
            <socketError>No route to host</socketError>
          </connect_ipv6_or_ipv4>
          <connect_ipv6_or_ipv4>
            <socketErrno>65</socketErrno>
            <socketError>No route to host</socketError>
          </connect_ipv6_or_ipv4>
          <connect_ipv6_or_ipv4>
            <socketErrno>65</socketErrno>
            <socketError>No route to host</socketError>
          </connect_ipv6_or_ipv4>
        </connectSocket>
        <error>Failed to establish initial TCP/IP connection</error>
        <hostname>192.168.86.152</hostname>
        <port>22</port>
      </sshConnect>
    </connectInner>
    <error>Failed.</error>
  </Connect_Ssh>
</WebSSHLog>
2022/01/24 00:48:05:342 [VERBOSE] Dismissing prompt...
2022/01/24 00:48:05:541 [VERBOSE] Host 192.168.86.69:22 changed. Is up : 0
2022/01/24 00:48:06:548 [VERBOSE] Host 192.168.86.152:22 changed. Is up : 0
2022/01/24 00:48:06:548 [VERBOSE] Updating of host statuses done.
2022/01/24 00:48:06:549 [VERBOSE] Need to refresh hosts list because at least one host status have been updated
2022/01/24 00:48:06:549 [VERBOSE] Fresh requested for connection list...
2022/01/24 00:48:07:387 [VERBOSE] Error : Can not establish connection to 192.168.86.152
2022/01/24 00:48:08:083 [VERBOSE] Updating host statuses...
2022/01/24 00:48:11:105 [VERBOSE] Updating of host statuses done.
2022/01/24 00:48:11:817 [DEBUG] Application will resign active [AppDelegate]
2022/01/24 00:48:12:449 [DEBUG] Application Did Enter Background
2022/01/24 00:48:51:222 [VERBOSE] Updating host statuses...
2022/01/24 00:48:51:668 [VERBOSE] Did become active [AppDelegate]
2022/01/24 00:48:52:989 [VERBOSE] Launching connection...
2022/01/24 00:48:52:991 [VERBOSE] Connecting to SSH...
2022/01/24 00:48:52:991 [VERBOSE] Launching connection with block...
2022/01/24 00:48:52:994 [VERBOSE] Won't dismiss prompt because no one is displayed.
2022/01/24 00:48:53:054 [VERBOSE] <WebSSHLog>
  <Connect_Ssh>
    <WebSSHVersion>9.5.0.88</WebSSHVersion>
    <connectInner>
      <hostname>192.168.86.152</hostname>
      <port>22</port>
      <sshConnect>
        <connectSocket>
          <connect_ipv6_or_ipv4>
            <socketErrno>65</socketErrno>
            <socketError>No route to host</socketError>
          </connect_ipv6_or_ipv4>
          <connect_ipv6_or_ipv4>
            <socketErrno>65</socketErrno>
            <socketError>No route to host</socketError>
          </connect_ipv6_or_ipv4>
          <connect_ipv6_or_ipv4>
            <socketErrno>65</socketErrno>
            <socketError>No route to host</socketError>
          </connect_ipv6_or_ipv4>
          <connect_ipv6_or_ipv4>
            <socketErrno>65</socketErrno>
            <socketError>No route to host</socketError>
          </connect_ipv6_or_ipv4>
        </connectSocket>
        <error>Failed to establish initial TCP/IP connection</error>
        <hostname>192.168.86.152</hostname>
        <port>22</port>
      </sshConnect>
    </connectInner>
    <error>Failed.</error>
  </Connect_Ssh>
</WebSSHLog>
2022/01/24 00:48:53:155 [VERBOSE] Dismissing prompt...
2022/01/24 00:48:54:241 [VERBOSE] Updating of host statuses done.
2022/01/24 00:48:55:236 [VERBOSE] Error : Can not establish connection to 192.168.86.152
2022/01/24 00:48:57:926 [DEBUG] Application will resign active [AppDelegate]
2022/01/24 00:48:58:565 [DEBUG] Application Did Enter Background

**Reply by isontheline** [reply]
@RonaldTHerman It means that your device isn't on the same subnet as your RPI. 

What's your iDevice IP address? 
You could extract it from mashREPL inside WebSSH.

**Comment by RonaldTHerman** [comment]
Still not getting it

I’m on an iPad iOS version 15.3 and trying to connect to a raspberry pi at 192.168.86.152:22.  Here’s what I get

01/27 13:32:13:604 [INFO] Opening embedded browser on : https://webssh.net/support/?ts=202201271332
2022/01/27 13:32:56:252 [VERBOSE] IAP already purchased : com.webssh.pro.unlimited
2022/01/27 13:33:00:594 [VERBOSE] Updating host statuses...
2022/01/27 13:33:03:605 [VERBOSE] Updating of host statuses done.
2022/01/27 13:33:16:223 [VERBOSE] Launching connection...
2022/01/27 13:33:16:225 [VERBOSE] Connecting to SSH...
2022/01/27 13:33:16:225 [VERBOSE] Launching connection with block...
2022/01/27 13:33:16:229 [VERBOSE] Won't dismiss prompt because no one is displayed.
2022/01/27 13:33:16:285 [VERBOSE] <WebSSHLog>
  <Connect_Ssh>
    <WebSSHVersion>9.5.0.88</WebSSHVersion>
    <connectInner>
      <hostname>192.168.86.152</hostname>
      <port>22</port>
      <sshConnect>
        <connectSocket>
          <connect_ipv6_or_ipv4>
            <socketErrno>65</socketErrno>
            <socketError>No route to host</socketError>
          </connect_ipv6_or_ipv4>
          <connect_ipv6_or_ipv4>
            <socketErrno>65</socketErrno>
            <socketError>No route to host</socketError>
          </connect_ipv6_or_ipv4>
          <connect_ipv6_or_ipv4>
            <socketErrno>65</socketErrno>
            <socketError>No route to host</socketError>
          </connect_ipv6_or_ipv4>
          <connect_ipv6_or_ipv4>
            <socketErrno>65</socketErrno>
            <socketError>No route to host</socketError>
          </connect_ipv6_or_ipv4>
        </connectSocket>
        <error>Failed to establish initial TCP/IP connection</error>
        <hostname>192.168.86.152</hostname>
        <port>22</port>
      </sshConnect>
    </connectInner>
    <error>Failed.</error>
  </Connect_Ssh>
</WebSSHLog>
2022/01/27 13:33:16:396 [VERBOSE] Dismissing prompt...
2022/01/27 13:33:18:469 [VERBOSE] Error : Can not establish connection to 192.168.86.152
2022/01/27 13:33:20:596 [VERBOSE] Updating host statuses...
2022/01/27 13:33:23:614 [VERBOSE] Updating of host statuses done.
2022/01/27 13:33:27:407 [VERBOSE] Updating host statuses...
2022/01/27 13:33:30:418 [VERBOSE] Updating of host statuses done.
2022/01/27 13:33:48:008 [INFO] Opening embedded browser on : https://webssh.net/documentation/help/networking/port-knocking/?ts=202201271333
2022/01/27 13:34:39:737 [VERBOSE] Updating host statuses...
2022/01/27 13:34:42:749 [VERBOSE] Updating of host statuses done.
2022/01/27 13:34:47:965 [DEBUG] Application will resign active [AppDelegate]
2022/01/27 13:34:48:611 [DEBUG] Application Did Enter Background


Same thing if a connect to macOS Monterey at 192.168.86.67. But your app on the macOS does connect to the rpi. So iPad must be at fault
---------------------------------
Ronald T. Herman
***@***.***
Canterbury NH

**Reply by isontheline** [reply]
Could you post the IP address of your iPad?

WebSSH > mashREPL > `ifconfig | grep inet`

**Comment by RonaldTHerman** [comment]
inet 192.168.86.249 netmask 0xffffff00 broadcast 192.168.86.255

---------------------------------
Ronald T. Herman
***@***.***
Canterbury NH

**Reply by isontheline** [reply]
Strange :/

Does a reboot helps?

**Comment by RonaldTHerman** [comment]
Nope.  A week ago I could connect to my macOS system. Now I can’t. I tried deleting the known_hosts file in ~/.ssh on the pi. Didn’t make a difference. As I said my macOS system can connect to the pi with WebSSH so it must be the pad, maybe something changed with the latest update to 15.3??

---------------------------------
Ronald T. Herman
***@***.***
Canterbury NH

**Reply by isontheline** [reply]
I'm downloading 15.3 in order to give it a try

**Reply by isontheline** [reply]
Works on iPhone Xr 15.3
Now trying iPad mini 4

**Reply by isontheline** [reply]
Also OK on iPad

Please check this iOS setting :
iOS > Settings > WebSSH > Local Network

It must be enabled in order to let you access to your local network

**Comment by isontheline** [comment, marked as answer]
Please check this iOS setting :
iOS > Settings > WebSSH > Local Network

It must be enabled in order to let you access to your local network

**Comment by RonaldTHerman** [comment]
Bingo!  Don’t know how that got changed. 

Thanks!

**Reply by isontheline** [reply]
You're Welcome!
